src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicListener.cs

```diff
@@ -183,19 +180,26 @@ private unsafe IPEndPoint Start(QuicListenerOptions options)
 
             Internals.SocketAddress address = IPEndPointExtensions.Serialize(listenEndPoint);
 
-            uint status;
-
             Debug.Assert(_stateHandle.IsAllocated);
 
             MemoryHandle[]? handles = null;
-            QuicBuffer[]? buffers = null;
+            // TODO: should they be pinned? The content is pinned, but the array holding the individual buffers isn't.
+            QUIC_BUFFER[]? buffers = null;
             try
             {
                 Debug.Assert(!Monitor.IsEntered(_state), "!Monitor.IsEntered(_state)");
                 MsQuicAlpnHelper.Prepare(applicationProtocols, out handles, out buffers);
+                // TODO: is the layout same for SocketAddress.Buffer and QuicAddr?
```

Comment:
IIRC, SocketAddress.Buffer does not take into account macOS having the family in a different location then linux. But, QuicAddr is identical to sockaddr, so theoretically if there is an internal SocketAddress to sockaddr helper, that would work fine.